### ISSUE TITLE
Testing playbook fails to run

### ISSUE THREAD

**Issue by lioramilbaum** [openshift/contra-hdsl#68]
The error:
"msg": "Failed to find required executable virtualenv in paths: /ansible//bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"

Full log file:
https://jenkins-ember-csi.apps.ci.centos.org/job/PR_submitted_CI/job/ci-automation-1/135/console

playbook:
https://github.com/lioramilbaum/ember-csi/blob/ci-automation-1/ci-automation/playbooks/tests/flake8.yml

**Comment by lioramilbaum** [comment]
When running a local container (on my laptop) with the same base image (ansible-executor), there is /usr/bin/virtualenv

**Comment by lioramilbaum** [comment]
The playbook sometime works. For example:
https://jenkins-ember-csi.apps.ci.centos.org/job/PR_submitted_CI/job/ci-automation-1/118/

**Comment by robnester-rh** [comment]
Please rerun with:

`executeTests verbose: true`

And post the full log.

**Comment by lioramilbaum** [comment]
Here is the Jenkinsfile snippet I have been using:

    stage("Execute Tests"){
      try {
        executeTests verbose: true,
                     vars: [ workspace: "${WORKSPACE}" ],
                     ansibleContainerName: "ansible-executor"
      } finally {
        junit 'junit.xml'
      }
    }

**Comment by robnester-rh** [comment]
Additionally, looking at the console output from your two runs, I see the following:

build 118, passing:

``` console
+ ansible-playbook -vvv -i /workDir/workspace/PR_submitted_CI_ci-automation-1/inventory /workDir/workspace/PR_submitted_CI_ci-automation-1/ci-automation/playbooks/tests/flake8.yml -e workspace=/workDir/workspace/PR_submitted_CI_ci-automation-1
ansible-playbook 2.6.3
```

build 140, failing:
``` console
+ ansible-playbook -vvv -i /workDir/workspace/PR_submitted_CI_ci-automation-1/inventory /workDir/workspace/PR_submitted_CI_ci-automation-1/ci-automation/playbooks/tests/flake8.yml -e workspace=/workDir/workspace/PR_submitted_CI_ci-automation-1
ansible-playbook 2.5.2
```

Please note the ansible-playbook versions. I'm curious as to why they would be different. They seem to change around build 120 and regress from 2.6.3 to 2.5.2, which also seems to correlate to continued failures?

**Comment by lioramilbaum** [comment]
@robnester-rh strange indeed

**Comment by lioramilbaum** [comment]
BTW, I have rebuilt today the ansible-executor image as soon as I started hitting this issue

**Comment by robnester-rh** [comment]
can you paste the s2i template used to rebuild the ansible-executor?

**Comment by lioramilbaum** [comment]
Where can I find it?
I have done the rebuilt from the OCP web console.  Went to the Builds->ansible-executor and used the 'start build' button

**Comment by robnester-rh** [comment]
please perform the following: `oc describe template ansible-executor`

**Comment by lioramilbaum** [comment]
@robneapiVersion: template.openshift.io/v1
kind: Template
labels:
  template: ansible-executor
metadata:
  creationTimestamp: '2018-10-07T12:15:21Z'
  name: ansible-executor
  namespace: ember-csi
  resourceVersion: '80744672'
  selfLink: >-
    /apis/template.openshift.io/v1/namespaces/ember-csi/templates/ansible-executor
  uid: a9495b5a-ca2a-11e8-80c2-0cc47a66a374
objects:
  - apiVersion: v1
    kind: ImageStream
    metadata:
      annotations:
        description: A container used to execute ansible commands
        iconClass: ansible
        openshift.io/display-name: Ansible Executor (Ephemeral)
        openshift.io/provider-display-name: 'Red Hat, Inc.'
        tags: ansible
      name: ansible-executor
    spec: {}
  - apiVersion: v1
    kind: BuildConfig
    metadata:
      name: ansible-executor
    spec:
      output:
        to:
          kind: ImageStreamTag
          name: 'ansible-executor:latest'
      source:
        dockerfile: 'FROM contrainfra/ansible-executor:v1.1.2'
        type: DockerFile
      strategy:
        dockerStrategy:
          env:
            - name: CONTAINER_NAME
              value: ansible-executor
            - name: GIT_SSL_NO_VERIFY
              value: 'true'
          noCache: true
        type: Docker
      triggers:
        - type: ConfigChange
ster-rh

**Comment by robnester-rh** [comment]
could you please execute:
`oc login`
`oc project <your project namespace>`
`oc describe template ansible-executor `

and paste the output here?

**Comment by lioramilbaum** [comment]
[lmilbaum@lmilbaum ember-csi]$ ~/Downloads/openshift-origin-server-v3.7.2-282e43f-linux-64bit/oc describe template ansible-executor
Name:		ansible-executor
Namespace:	ember-csi
Created:	8 weeks ago
Labels:		<none>
Annotations:	<none>

Parameters:	 

Object Labels:	template=ansible-executor

Message:	<none>

Objects:	 
    ImageStream	ansible-executor
    BuildConfig	ansible-executor

**Comment by arilivigni** [comment]
@lioramilbaum @robnester-rh first off this 100% NOT an contra-hdsl issue.

Rob and I tried several experiments and we were able to get to the point that locally running this on OpenShift environments things worked fine but fail in the ember-csi OpenShift project.

We ran a pipeline in ember-csi and it could not find the virtualenv.

1. In the ember-csi project we deleted the ansible-executor buildconfig, image, and template
2. Added the template back rebuilt and tagged the image as stable

This still resulted in the virtualenv failing to be found

Tonight I then went to the continuous-infra project namespace and saw there was an ansible-executor there as well that was old.

I repeated the above 2 steps then ran the same pipeline and it worked.  I then went back to ember-csi and ran the same pipeline it also worked.

In conclusion I believe there is something that was pulling a stale image from a different namespace.  Why I have no idea and maybe it is a bug.  This is also an older OpenShift 3.6 cluster.  Maybe a question for @bstinsonmhk.

Rob is working on an enhancement to the H-DSL that will pull down the correct image versions from Dockerhub that may have avoided this but given this may be an OpenShift issue I am not 100% sure.

In either case if you ran this locally on your minishift environment it would be clear that this had nothing to do with contra-hdsl.  While we are happy to help I hope over time you and your team will grow to be able to debug OpenShift and Jenkins issues.
